metron-interface/metron-alerts/src/app/alerts/alerts-list/table-view/table-view.component.ts

```diff
@@ -77,12 +80,24 @@ export class TableViewComponent implements OnChanges {
               searchService: SearchService,
               metronDialogBox: MetronDialogBox,
               updateService: UpdateService,
-              metaAlertService: MetaAlertService) {
+              metaAlertService: MetaAlertService,
+              globalConfigService: GlobalConfigService) {
     this.router = router;
     this.searchService = searchService;
     this.metronDialogBox = metronDialogBox;
     this.updateService = updateService;
     this.metaAlertService = metaAlertService;
+    this.globalConfigService = globalConfigService;
+  }
+
+  ngOnInit() {
+    this.globalConfigService.get().subscribe((config: {}) => {
+      this.globalConfig = config;
+      if (this.globalConfig['source.type.field'] === 'source.type' && !this.alertsColumnsToDisplay['source.type']) {
```

Comment:
I don't think this logic is correct.  What happens if you have "source.type" selected as a table column?  I went in a updated by local storage directly and ended up with 2 "source.type" columns.

Reply:
@merrimanr Good point. What if we check if a config property of ['source.type.field'] exists and change it to whatever value it has. If no config exists, default to `source:type` (to catch existing users without a config property set)?

```
if (this.globalConfig['source.type.field']) {
        this.alertsColumnsToDisplay.splice(2, 0, new ColumnMetadata(config['source.type.field'], 'string'));
      } 
else {
        this.alertsColumnsToDisplay.splice(2, 0, new ColumnMetadata(config['source:type'], 'string'));
      }
```